### PR TITLE
WIP X9 openapi yaml used to generate HTTP API

### LINKED ISSUE #37: Create openapi.yaml
Create openapi.yaml

### PR DESCRIPTION
**Pull request by bkmoovio**
Closes #37

### CODE REVIEW

**Comment by bkmoovio** [comment]
Take a look fellas and let me know what I need to adjust!

File: checkDetail.go [hunk 1]
```diff
@@ -303,11 +303,15 @@ func (cd *CheckDetail) fieldInclusion() error {
 	return nil
 }
 
+// ToDo:NBSM?
```
[the remark this answers is not in the document]
Comment (bkmoovio) [reply checkDetail.go]: Will do

File: client/git_push.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
```
[the remark this answers is not in the document]
Comment (bkmoovio) [reply client/git_push.sh]: It's a generated file via openapi-generator-cli so I will have to research that.

File: converters.go [hunk 3]
```diff
@@ -73,6 +73,8 @@ func (c *converters) numericField(n int, max uint) string {
 	return s
 }
 
+// ToDo: Review with AuxiliaryOnUs and ExternalProcessingCode
```
[the remark this answers is not in the document]
Comment (bkmoovio) [reply converters.go]: Will add

File: .travis.yml [hunk 4]
```diff
@@ -11,22 +11,21 @@ matrix:
...
 script:
 # Just check gofmt on linux, it's the fastest builder
 - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then test -z $(gofmt -s -l $GOFILES); fi
-- go test ./... -race -coverprofile=coverage.txt -covermode=atomic
+# - go test ./... -race -coverprofile=coverage.txt -covermode=atomic
 - misspell -error -locale US $GOFILES
 - gocyclo -over 50 $GOFILES
 - golint -set_exit_status $GOFILES
-- staticcheck ./...
+- staticcheck ./cmd/*/*.go ./pkg/*/*.go *.go
```
Comment (adamdecaf) [inline .travis.yml]: There's no `./pkg/*/*.go` directory in this project.

File: openapi.yaml [hunk 5]
```diff
@@ -0,0 +1,3036 @@
...
+  description: Moov X9 () implements an HTTP API for creating, parsing and validating X9 (Check21) files.
+  version: v1
+  title: X9 API
+  contact:
+    url: https://github.com/moov-io/x9
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+servers:
+  - url: http://localhost:8083
```
Comment (adamdecaf) [inline openapi.yaml]: I created a PR to declare `wire` on port `:8088` - https://github.com/moov-io/base/pull/45

File: openapi.yaml [hunk 6]
```diff
@@ -0,0 +1,3036 @@
...
+    description: Local development
+
+tags:
+  - name: Files
```
Comment (adamdecaf) [inline openapi.yaml]: Can we call this tag Wire? `Files` is already taken by ACH files in our [api docs](https://api.moov.io/#tag/Files).

File: client/README.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
+# Go API client for openapi
+
+Moov X9 () implements an HTTP API for creating, parsing and validating X9 (Check21) files.
```
[the remark this answers is not in the document]
Comment (bkmoovio) [reply client/README.md]: Yes.

File: .travis.yml (hunk 4 above)
Comment (bkmoovio) [reply .travis.yml]: That is correct we do not have a ./pkg/*/*.go directory in this project yet

File: openapi.yaml (hunk 5 above)
Comment (bkmoovio) [reply openapi.yaml]: Thanks

File: openapi.yaml (hunk 6 above)
Comment (bkmoovio) [reply openapi.yaml]: Adam and I had a slack chat, we are going to go with ACH Files, X9 Files, and Wire Files

**Pull request closed without merging**